pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -261,12 +277,36 @@ func ResolvePipelineRun(
 		if taskRun != nil {
 			rprt.TaskRun = taskRun
 		}
+
+		// Get all conditions that this pipelineTask will be using, if any
+		if len(pt.Conditions) > 0 {
+			rcc, err := resolveConditionChecks(&pt, pipelineRun.Status.TaskRuns, rprt.TaskRunName, getTaskRun, getCondition)
+			if err != nil {
+				return nil, err
+			}
+			rprt.ResolvedConditionChecks = rcc
+		}
+
 		// Add this task to the state of the PipelineRun
 		state = append(state, &rprt)
 	}
 	return state, nil
 }
 
+// getConditionCheckName should return a unique name for a `ConditionCheck` if one has not already been defined, and the existing one otherwise.
+func getConditionCheckName(taskRunStatus map[string]*v1alpha1.PipelineRunTaskRunStatus, trName, conditionName string) string {
+	trStatus, ok := taskRunStatus[trName]
+	if ok && trStatus.ConditionChecks != nil {
+		for k, v := range trStatus.ConditionChecks {
+			// TODO: We should allow multiple conditions of the same Name type?
```

Comment:
hm do you mean something like this:

```
conditions:
  - conditionRef: "always-true"
  - conditionRef: "always-true"
```

It's hard to imagine needing this so my inclination is to suggest we don't allow it until it becomes clear we need it?

Reply:
We might need it for something like two `stringEquals` conditions together. Though its definitely more of an advanced use case we can tackle in https://github.com/tektoncd/pipeline/issues/1022